File: src/content/3.9/code/scala/snippet09.scala
implicit def storeFunctor[S] = new Functor[Store[S, ?]] {
  def fmap[A, B](g: A => B)(fa: Store[S, A]): Store[S, B] = fa match {
    case Store(f, s) =>
      Store(g compose f, s)
  }
}

File: src/content/3.9/code/ocaml/snippet09.ml
module Store_Functor (S : sig
  type s
end) : Functor with type 'a t = (S.s, 'a) store = struct
  type 'a w = (S.s, 'a) store
  type 'a t = 'a w

  let fmap g (Store (f, s)) = Store (Fun.compose g f, s)
end
